### DISCUSSION TITLE
Environment variables for database setup in container

### DISCUSSION THREAD

**Discussion by btrepp** [smallstep/certificates#1428] [Ideas]
Hi there,

First of, this tool is amazing, an ACME provider that I can easily manage private certificates, its a great missing piece for hobby labs!.

I've been integrating it with my k8s setup, and made some of the manifests/kustomizations https://git.sr.ht/~btrepp/environment/tree/main/item/dist/step-ca/latest/base

One of the issues I've encountered, is there doesn't seem to be an easy way to container the docker container to use a database instead of the badger DB. My motivation to do this is that I already have some postgres bases that use rsnapshot to ensure I have backups, at the moment with a StatefulSet/PVC provisioner it works okay, but its fairly easy to 'remove' your pvc and potentially lose your data :).

It would be great if the docker container could be pointed to databases as env vars (I've tried to look at the entry point, and it doesn't look like this is supported yet). That way it's easy to keep the data backed up if needed, plus also easy to run the step-ca as a deployment that could scale, and leave postgres as a stateful set for persistence.

**Comment by tashian** [comment]
Hi, yes, you're right. One issue here is that our `step ca init` command (which is run by the entrypoint script if no PKI exists) doesn't allow configuration of the database connection, so we don't have an easy way to do this through env vars. So, you won't be able to use our automation entrypoint script for this. What you'll need to do instead is put together a `ca.json` file and inject or mount it into the container.

And, you're welcome to open a feature request issue on this repo for this, if it's something you'd like us to consider adding.

**Reply by mikaelparkefelt** [reply]
I need this function as well....

Why haven't you done this very important function already?

It is the key for me to automation to work properly....
Could you please supply us with the minimum ca.json that is needed to make step ca init operate on the external database instead?
